private/rules/javadoc.bzl

```diff
@@ -33,15 +34,27 @@ def generate_javadoc(
     args.add("--element-list", element_list)
     args.add_all(source_jars, before_each = "--in")
     inputs.extend(source_jars)
+    args.add_all(ctx.files.doc_resources, before_each = "--resources")
+    inputs.extend(ctx.files.doc_resources)
     args.add_all(classpath, before_each = "--cp")
     transitive_inputs.append(classpath)
+
     for dep in doc_deps:
         dep_info = dep[_JavadocInfo]
         args.add("-linkoffline")
         args.add(dep_info.url)
         args.add(dep_info.element_list.dirname)
         inputs.append(dep_info.element_list)
-    args.add_all(javadocopts)
+
+    processed_opts = []
+    for opt in javadocopts:
+        if "$(" in opt:
+            processed_opt = ctx.expand_make_variables("javadocopts", opt, ctx.var)
```

Comment:
Is there any harm in just expanding the make variables unconditionally? In the common case, I'd expect it to be a no-op, and it makes the code a little easier to read.

Reply:
Seems to work!